webui/src/pages/auth/login.jsx

```diff
@@ -23,18 +23,16 @@ const LoginForm = () => {
                 <Card className="login-widget">
                     <Card.Header>Login</Card.Header>
                     <Card.Body>
-                        <Form onSubmit={(e) => {
+                        <Form onSubmit={async (e) => {
                             e.preventDefault()
-                            auth
-                                .login(e.target.username.value, e.target.password.value)
-                                .then(user => {
-                                    setLoginError(null);
-                                    return router.push((!!next) ? next : '/');
-                                })
-                                .catch(err => {
-                                    setLoginError(err);
-                                })
-
+                            try {
+                                await auth.login(e.target.username.value, e.target.password.value);
+                                setLoginError(null);
+                                const nextOrRoot = next && next !== router.route ? next : '/';
```

Comment:
Not a fan of this change, then.  In general I do not trust setting a known-incorrect value of `next` and then attempting to fix it at the last moment; I would much prefer to set a correct value.

To amplify on my previous points:

## 1. Where is the special treatment of a falsy `next`?

Before the change it would redirect to `/`, now it redirects to whatever the falsy value does.

## 2. Let's set the correct value!

## 3. Behaviour on JWT timeout

I believe that this does the wrong thing whenever I am logged out, e.g. (also) if my cookie times out.

I tried this experiment:
1. Log in.
2. Navigate to repos page.
3. Erase the token cookie.
4. Click a repo, get redirected to login page (expected behaviour).
5. ... login page has a bad `next`.

With this patch, the login page will _still_ have  a bad `next`, and my reading of the fix above is that it will bring me to the main page.  The **expected** behaviour is to redirect to the old page after logging in -- AFAICT this patch does not do that.

Reply:
Moved the check before we set the value of 'next' when route to the login page.

Reply:
Moved the check before we set the value of 'next' when route to the login page.